### PR TITLE
Feature/cu 1fmhxg2 Fanc EMP & Onimaru PUSH feedback Enemy (DONE)

### PR DESCRIPTION
**Pull request by telmiuse**

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-1fmhxg2 EMP Recived Effect](https://app.clickup.com/t/1fmhxg2)

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp [hunk 1]
```diff
@@ -405,6 +423,7 @@ void AIMeleeGrunt::OnCollision(GameObject& collidedWith, float3 collisionNormal,
 					if (animation->GetCurrentState()) {
 						animation->SendTrigger(animation->GetCurrentState()->name + "StunStart");
 					}
+					particlesEmp->PlayChildParticles();
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/AIMeleeGrunt.cpp:426]: Check nullity. In the start you check the nullity of the aux local variable, but at this point this pointer could be null

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp [hunk 2]
```diff
@@ -418,6 +437,7 @@ void AIMeleeGrunt::EnableBlastPushBack() {
 	if (state != AIState::START && state != AIState::SPAWN && state != AIState::DEATH) {
 		gruntCharacter.beingPushed = true;
 		state = AIState::PUSHED;
+		particlesPush->PlayChildParticles();
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/AIMeleeGrunt.cpp:440]: Check nullity. In the start you check the nullity of the aux local variable, but at this point this pointer could be null

File: Penteract/Assets/Scripts/RangedAI.cpp [hunk 3]
```diff
@@ -607,6 +630,7 @@ void RangedAI::PlayAudio(AudioType audioType) {
 void RangedAI::EnableBlastPushBack() {
 	if (state != AIState::START && state != AIState::SPAWN && state != AIState::DEATH) {
 		ChangeState(AIState::PUSHED);
+		particlesPush->PlayChildParticles();
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/RangedAI.cpp:633]: Check nullity. In the start you check the nullity of the aux local variable, but at this point this pointer could be null

File: Penteract/Assets/Scripts/RangedAI.cpp [hunk 4]
```diff
@@ -275,7 +295,10 @@ void RangedAI::OnCollision(GameObject& collidedWith, float3 collisionNormal, flo
 			if (collidedWith.name == "EMP") {
 				if (agent) agent->RemoveAgentFromCrowd();
 				stunTimeRemaining = stunDuration;
-				if (state != AIState::STUNNED) ChangeState(AIState::STUNNED);
+				if (state != AIState::STUNNED) {
+					ChangeState(AIState::STUNNED);
+					particlesEmp->PlayChildParticles();
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/RangedAI.cpp:300]: Check nullity. In the start you check the nullity of the aux local variable, but at this point this pointer could be null

**Review by Jordi-Gil — CHANGES_REQUESTED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Pull request merged**